### PR TITLE
feat: support None to remove envvars

### PR DESCRIPTION
**Pull request by henryiii**
Followup to #804. Setting `env={"NOCOLOR": None}` will unset the value if it exists in the outer environment.

### CODE REVIEW

**Comment by henryiii** [comment]
I got a thumbs up, anyone want to do a review? Nice to have an approval. ;)

**Review by cjolowicz — APPROVED** [review]

**Comment by edgarrmondragon** [comment]
Should there be a documentation update for this?

**Comment by henryiii** [comment]
I did add it to the docs. It's only present in nox/sessions.py, but that's the only place `env=` is documented. It shows up in the docs page, though.

File: nox/sessions.py [hunk 1]
```diff
@@ -350,7 +350,9 @@ def run(
             print("Current Git commit is", out.strip())
 
         :param env: A dictionary of environment variables to expose to the
-            command. By default, all environment variables are passed.
+            command. By default, all environment variables are passed. You
+            can block an environment variable from the outer environment by
+            setting it to None.
```
Comment (henryiii) [inline nox/sessions.py:355]: Docs update here.

Comment (edgarrmondragon) [reply nox/sessions.py:355]: Ah, I do see it in https://nox.thea.codes/en/stable/config.html#nox.sessions.Session.run. Sorry about the noise.

**Pull request merged**